### PR TITLE
Fix various project + static analyzer warnings + other fixes

### PR DESCRIPTION
**Pull request by jessesquires**
### Description

- Fix implicit numeric type conversion warnings by making them explicit
- Prefer checking `.isEmpty` rather than `.count == 0` for Swift collections (can be more optimal)
- Avoid comparing `BOOL` explicitly to `NO` (considered bad practice since `BOOL` is a `signed char`)
- Specify unused `SGX_FLAGS` as `__unused` to silence warnings
- Fix some mismatched nullability annotations

### Contributor checklist

- [x] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [x] My commits are rebased on the latest master branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * Build only

### CODE REVIEW

File: Signal/src/call/OWSWebRTCDataProtos.pb.m [hunk 1]
```diff
@@ -107,13 +107,13 @@ - (OWSWebRTCProtosConnectedBuilder*) toBuilder {
 }
 - (void) writeDescriptionTo:(NSMutableString*) output withIndent:(NSString*) indent {
   if (self.hasId) {
-    [output appendFormat:@"%@%@: %@\n", indent, @"id", [NSNumber numberWithLongLong:self.id]];
+    [output appendFormat:@"%@%@: %@\n", indent, @"id", [NSNumber numberWithLongLong:(long long)self.id]];
```
Comment (charlesmchen-signal) [inline Signal/src/call/OWSWebRTCDataProtos.pb.m]: This is generated code.

Comment (jessesquires) [reply Signal/src/call/OWSWebRTCDataProtos.pb.m]: Ugh. Didn't realize. Now I notice the `.pb.m` 🤦‍♂️ 

Will revert.

File: Signal/src/environment/SignalApp.m [hunk 2]
```diff
@@ -137,7 +137,7 @@ - (NotificationsManager *)notificationsManager
     return _notificationsManager;
 }
 
-- (AccountManager *)accountManager
+- (nonnull AccountManager *)accountManager
```
Comment (michaelkirk-signal) [inline Signal/src/environment/SignalApp.m]: Hmmm, why is this necessary? 

Looks like the whole .m is within `NS_ASSUME_NONNULL_BEGIN`

Comment (jessesquires) [reply Signal/src/environment/SignalApp.m]: Ah, looks like this was a false positive (or dirty workspace). Will fix. 👌

**Comment by jessesquires** [comment]
updated 😄

**Comment by michaelkirk-signal** [comment]
In a2c65e3a4, thanks!
